### ISSUE TITLE
time.h is missing in llama.cpp

### ISSUE THREAD

**Issue by edp1096** [ggerganov/llama.cpp#907]
Hello folks.

I tried to compile using MinGW on windows and got following error message.

It was compiled well by adding `#include <time.h>` to `llama.cpp`.

I'm not familiar with c/cpp.
Is there something I lost?

```powershell
$ mingw32-make.exe
I llama.cpp build info: 
I UNAME_S:  
I UNAME_P:  
I UNAME_M:  

...

llama.cpp: In function 'llama_context* llama_init_from_file(const char*, llama_context_params)':
llama.cpp:1684:23: error: 'time' was not declared in this scope
         params.seed = time(NULL);
                       ^~~~
llama.cpp:1684:23: note: suggested alternative: 'ftime'
         params.seed = time(NULL);
                       ^~~~
                       ftime
mingw32-make: *** [Makefile:148: llama.o] Error 1

...

```

**Comment by edp1096** [comment]
I found the error caused from disappeared `<regex>` definition in `/llama.cpp` the commit after `aaf3b23`

Am I the only one who met this error?

**Comment by wro52** [comment]
No, not really. I am using Code::Blocks with TDM-GCC-64 and time is missing. By adding #include <time.h> in llama.cpp
everything works perfekt. With Code::Blocks it is very easy and fast to create a projekt and include the source files into that
project. Set up the compiler and linker and here it goes. Everything is compact, easy to store and forward. No further
dependancies like DLL's and one file does it all. By the way, fastest performance is with vicuna ggml-vicuna-7b-4bit.bin and
ggml-vicuna-7b-4bit-rev1.bin and this settings Intel core i5
following compiler / linker flags 
-fopenmp      (compiler/linker) 
-pthread       (compiler/linker)
-mmmx
-msse3
-mavx
-mf16c
-m models/vicuna/ggml-vicuna-7b-4bit-rev1.bin --color --threads 4 --batch_size 256 --n_predict -1 --top_k 12 --top_p 1 --temp 0.36 --repeat_penalty 1.05 --ctx_size 2048 --instruct --reverse-prompt "### Human:" -f prompts/chat-with-vicuna.txt

**Comment by edp1096** [comment]
The problem cause from MinGW gcc 8.1.0

Solved after changed MinGW version to 12.0.
